src/object/object_primitive.c

```diff
@@ -11990,7 +11990,14 @@ mr_cmpval_char (DB_VALUE * value1, DB_VALUE * value2, int do_coercion, int total
 
   if (!ignore_trailing_space)
     {
-      ti = (value1->domain.char_info.type == DB_TYPE_CHAR && value2->domain.char_info.type == DB_TYPE_CHAR);
+      if (do_coercion == 2)
+	{
+	  ti = (value1->domain.char_info.type == DB_TYPE_CHAR || value2->domain.char_info.type == DB_TYPE_CHAR);
+	}
+      else
+	{
+	  ti = (value1->domain.char_info.type == DB_TYPE_CHAR && value2->domain.char_info.type == DB_TYPE_CHAR);
+	}
```

Comment:
Can I know why both ignore_trailing_space (from getting parameter) and do_coercion need to be checked here? Are there any cases that do_coercion is given 2 but ignore_trailing_space is set to true at the parameter?

Reply:
At the "btree_get_prefix_separator", the btree_compare_key (finally called as char-type comparing) is called with mixed string argument (CHAR-type and VARCHAR-type).
If the key is composed of CHAR-type, the mixed comparison argument mis-operated as varchar-type-compare because the char-type comparison is operated only if both of two arguments are char-type.